03week/towersOfHanoi.js

```diff
@@ -19,24 +19,82 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+// moves the piece from the first stack to the second stack
+const movePiece = (input1, input2) => {
+  return stacks[input2].push(stacks[input1].pop(input1));
 }
 
-function isLegal() {
-  // Your code here
+// return true if both inputs (combined into an array) contain only expected values ['a', 'b', 'c']
+// Note: values have already been converted to lowercase and trimmed of spaces
+const isValid = (input1, input2) => {
+  if (input1 && input2) {
+    const combinedInputs = input1 += input2;
+    const combinedInputsToArray = (input1 += input2).split('');
+    return combinedInputsToArray.every(input => ['a', 'b', 'c'].includes(input));
+  }
+}
 
+// return true if the selected move is allowed (i.e. follows the rules of the game)
+const isLegal = (input1, input2) => {
+  return stacks[input1].length > 0 &&
+    stacks[input1].slice(-1) < stacks[input2].slice(-1) ||
+    stacks[input2].length === 0 &&
+    input1 !== input2
 }
 
-function checkForWin() {
-  // Your code here
+// return true if stack 'b' or 'c' have an array length of 4
+const checkForWin = () => {
+  return stacks.b.length === 4 || stacks.c.length === 4;
+}
 
+// return true if either input contains 'reset'
+const checkForReset = (input1, input2) => {
+  const combinedInputsToCheckForReset = input1 += input2;
```

Comment:
why are you reassigning what input1 is? best practice would be to make a new variable